### ISSUE TITLE
Delegate operator (by) produces nullpointer exeption on kotlin 1.8

### ISSUE THREAD

**Issue by JochenGuckSnk** [Kotlin/kotlinx.serialization#2157]
**Describe the bug**
after changing the project to kotlin version 1.8 a map delegate in a serializable class is no longer compiled.

Error:
Details: kotlinx.serialization compiler plugin internal error: unable to transform declaration, see cause
	at org.jetbrains.kotlinx.serialization.compiler.extensions.SerializerClassLowering.lower(SerializationLoweringExtension.kt:184)
...
Caused by: java.lang.NullPointerException
	at org.jetbrains.kotlinx.serialization.compiler.backend.ir.SerializableIrGenerator$generateInternalConstructor$1.invoke(SerializableIrGenerator.kt:137)

**To Reproduce**
```kotlin
@Serializable()
class Dto (
     val data: Map<Int, Int>
) : Map<Int, Int> by data
```

**Expected behavior**
It sould compile

**Environment**
 - Kotlin version: 1.8.0
 - Library version: 1.4.1
 - Kotlin platforms: Multiplatform
 - Gradle version: 7.4.2

[MinimumExample.zip](https://github.com/Kotlin/kotlinx.serialization/files/10424758/MinimumExample.zip)

**Comment by cowst** [comment]
It happens also with version 1.3.3 and gradle 7.5

**Comment by sandwwraith** [comment]
Fixed in Kotlin 1.8.10
